lldb/test/API/functionalities/tsan/swift/Makefile

```diff
@@ -1,6 +1,9 @@
 
 SWIFT_SOURCES := main.swift
 
+TSAN_DYLIB_DIR := $(CLANG_MODULE_CACHE_DIR)/../../../lib/lldb/swift/clang/lib/darwin/
```

Comment:
We need to find a way to not derive this from CLANG_MODULE_CACHE_DIR.

Reply:
Right - I dumped all variables available in this makefile, and LLVM_LIBS_DIR one seems better for this purpose. I'll update it, but want to see first what the discussion here is going to be like, perhaps we'll go some other way.

Reply:
The issue with using `LLVM_LIBS_DIR` is that it points to `$LLVM_BUILD_DIR/lib` instead of `$LLDB_BUILD_DIR/lib`. That means that `TSAN_DYLIB_DIR` would be `$(LLVM_LIBS_DIR)/clang/$clang_version/lib/darwin`, which would mean you would have to find and manage `$clang_version` yourself.

It might be worth modifying the test suite to make `LLDB_LIBS_DIR` available to the make rules so you can take advantage of that. That way we can use something like `$(LLDB_LIBS_DIR)/lldb/clang/lib/darwin` or something to that effect. What do you think?

Reply:
Another idea: If you know the compiler you are using, you could ask it for its resource directory with something like `$CXX -print-resource-dir` and then append `/lib/darwin` yourself. We do that elsewhere in the test suite. See `find_sanitizer_runtime` in `lldb/test/API/lit.cfg.py`.

Reply:
I noticed that LLDB_LIBS_DIR is actually set in lit.site.cfg.py.in, but not set into environment by lit.cfg.py - so I added it.

Reply:
> This idea seems good to me. A few questions:
> 1.) Will this need to be supported moving forward or just for this branch? If moving forward, you'll want to cherry-pick this to `next`.
> 2.) Is this specific to swift or does this need to happen for all TSAN/ASAN tests?

1.) I think it should be supported. We had previous issues with these tests failing to build on rebranch which this fix should address, to eliminate similar problems in the future. Yes, I will cherry-pick to next.
2.) These two tests (running ASANified/TSANinfied Swift code with LLDB API) are very specific, I don't currently know about any similar cases elsewhere.